references/cli/system.go

```diff
@@ -357,3 +394,27 @@ func NewSystemDiagnoseCommand(c common.Args) *cobra.Command {
 	}
 	return cmd
 }
+
+// CheckAPIService checks the APIService
+func CheckAPIService(ctx context.Context, config *rest.Config, apiService *apiregistrationV1beta.APIService) error {
+	svcName := apiService.Spec.Service.Name
+	svcNamespace := apiService.Spec.Service.Namespace
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	svc, err := clientset.CoreV1().Services(svcNamespace).Get(ctx, svcName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	set := labels.Set(svc.Spec.Selector)
+	listOptions := metav1.ListOptions{LabelSelector: set.AsSelector().String()}
+	pods, err := clientset.CoreV1().Pods(svcNamespace).List(ctx, listOptions)
```

Comment:
Here you are listing the cluster-gateway pods. It might be better if you add some additional checks for those pods. For example, check if these pods are ready.